test/test-parse-expression.cpp

```diff
@@ -1126,7 +1149,9 @@ TEST_F(test_parse_expression,
       auto guard = p.parser().enter_function(function_attributes::async);
       expression* ast = p.parse_expression();
       EXPECT_EQ(summarize(ast), test.expected_async_function);
-      EXPECT_THAT(p.errors(), IsEmpty());
+
+      EXPECT_THAT(p.errors(), ::testing::Not(::testing::Contains(DIAG_TYPE(
+                                  diag_await_operator_outside_async))));
```

Comment:
Must fix: Negative assertions are bad and we should avoid them if possible.

If the diagnostics change for only one test case, we should special-case that test case (similar to what we do on line 1126).